cartridges/int_algolia_sfra/cartridge/controllers/Search.js

```diff
@@ -97,7 +97,44 @@ server.replace('Show', cache.applyShortPromotionSensitiveCache, consentTracking.
                 cgid: req.querystring.cgid,
                 q: req.querystring.q,
                 activePromotions: activePromotions
-            });
+            }
+
+            var storeList = [];
+
+            var attributeList = algoliaData.getSetOfArray('AdditionalAttributes');
+            var isStoreAvailabilityEnabled = attributeList.indexOf('storeAvailability') !== -1;
+
+            if (isStoreAvailabilityEnabled) {
+                // Do not need to cache this, because page is already cached via cache.applyShortPromotionSensitiveCache
+                var StoreMgr = require('dw/catalog/StoreMgr');
+
+                // Get all stores using a very large radius search
+                var storesMap = StoreMgr.searchStoresByCoordinates(0, 0, 'mi', 99999999);
```

Comment:
Why using this method and not the same than in the job (`SystemObjectMgr.getAllSystemObjects('Store')`)?
It look more "hacky"

Reply:
Because there is no permission to use it on controllers. This method is only available for the jobs.
there is a [quota](https://salesforcecommercecloud.github.io/b2c-dev-doc/docs/current/quota/html/API_Quotas.html) for storefront requests
![Image2025-03-11 at 13 54 27](https://github.com/user-attachments/assets/67c1b31d-eaf4-49c0-b305-1968073bf780)